src/strategies/volt-voting-power/index.ts

```diff
@@ -0,0 +1,291 @@
+    '361': 'https://geyser-graph-on-theta.voltswap.finance/subgraphs/name/theta/geyser-v2'
+
+}
+
+
+const abi = [
+    {
+        constant: true,
+        inputs: [
+            {
+                internalType: 'address',
+                name: 'user',
+                type: 'address'
+            }
+        ],
+        name: 'balanceOf',
+        outputs: [
+            {
+                internalType: 'uint256',
+                name: '',
+                type: 'uint256'
+            }
+        ],
+        payable: false,
+        stateMutability: 'view',
+        type: 'function'
+    }
+];
+
+
+
+export async function strategy(
+    _space,
+    network,
+    _provider,
+    addresses,
+    options,
+    snapshot
+) {
+
+    const lpTokenAddress = options.lpTokenAddress.toLowerCase();
+    const tokenDecimals = options.tokenDecimals
+
+
+
+    const voltDataparams = {
+        users: {
+            __args: {
+                where: {
+                    id_in: addresses.map((address) => address.toLowerCase())
+                },
+                first: 1000
+            },
+            id: true,
+
+            vaults: {
+                id: true,
+                locks: {
+                    __args: {
+                        where: {
+                            token_in: [options.voltAddress.toLowerCase(), options.lpTokenAddress.toLowerCase()]
+                        }
+                    },
+                    id: true,
+                    token: true,
+                    amount: true,
+                    stakeUnits: true
+                }
+
+
+            }
+        }
+    }
+
+
+
+
+    const subgraphDataParams = {
+
+
+        pairs: {
+            __args: {
+                where: {
+                    id: lpTokenAddress.toLowerCase()
+                },
+                first: 1
+            },
+            id: true,
+            token0:
+            {
+                id: true
+            },
+            reserve0: true,
+            token1: {
+                id: true
+            },
+            reserve1: true,
+            totalSupply: true
+
+
+        }
+    };
+
+
+
+
+
+
+    const poolData = await subgraphRequest(SUBGRAPH_URL[options.network || network], voltDataparams)
+
+    const subgraphData = await subgraphRequest(UNISWAP_SUBGRAPH_URL[options.network || network], subgraphDataParams);
+
+
+    let totalVoltComposition = 0
+    let totalStake = 0
+
+
+    if (subgraphData && subgraphData.pairs) {
+
+        subgraphData.pairs.forEach((lp) => {
+            const isToken0 = lp.token0.id.toLowerCase() === options.voltAddress.toLowerCase()
+            if (isToken0) {
+                totalVoltComposition = lp.reserve0
+
+            } else {
+                totalVoltComposition = lp.reserve1
+            }
+            totalStake = Number(lp.reserve0) + Number(lp.reserve1)
+
+        });
+
+    }
+
+
+
+
+
+
+
+
+
+
+    if (addresses.length && addresses.length > 200) {
```

Comment:
I don't understand this 🤔 voting power should not be based on number of voters 🤔 maybe try paginating requests, if it is just `balanceOf` function you can call `erc20-balance-of strategy` directly from here, example:
https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/balance-of-with-min/index.ts#L14

Reply:
Yes. This is paginating the requests, however, it's not just the balanceOf that's needed but the sum of the user volt balance, staked volt balance and user LP share mapped volt balance.
The formula is : user's volt balance + staked volt balance + User LP share mapped volt balance

Reply:
> Yes. This is paginating the requests, however, it's not just the balanceOf that's needed but the sum of the user volt balance, staked volt balance and user LP share mapped volt balance.
> The formula is : user's volt balance + staked volt balance + User LP sh

but i see only balanceOf function in the multical and abi 🤔

Reply:
Yes, after getting the balance of the user, there's a call to the subgraphs to user's stake in the volt pool and also the user's share in the percentage of volts in the LP token. All these values are needed to calculate the user's votes.

Reply:
> Yes, after getting the balance of the user

Instead of calling multicall, is it possible to call erc20-balance-of strategy directly, like [here ](https://github.com/snapshot-labs/snapshot-strategies/blob/master/src/strategies/balance-of-with-min/index.ts#L14)

Reply:
oh I get your point now. Okay, I can call and use the  erc20BalanceOfStrategy. My bad. Let me fix that.

Reply:
I have done the changes now. I am making a call erc20-balance-of strategy directly instead of using the multicall. I have also resolved the issues with this PR removing some strategies. The package-lock.json changes have been reverted to the original.